NDTensors/src/lib/Sectors/src/abstractcategory.jl

```diff
@@ -1,45 +1,115 @@
+function Base.isless(c1::C, c2::C) where {C<:AbstractCategory}
+  return isless(category_label(c1), category_label(c2))
+end
 
-function dimension(c::AbstractCategory)
-  return error("method `dimension` not defined for type $(typeof(c))")
+# =================================  Sectors interface  ====================================
+trivial(x) = trivial(typeof(x))
+function trivial(axis_type::Type{<:AbstractUnitRange})
+  return gradedrange([trivial(eltype(axis_type))])  # always returns nondual
+end
+function trivial(la_type::Type{<:LabelledInteger})
+  return labelled(one(unlabel_type(la_type)), trivial(label_type(la_type)))
+end
+function trivial(type::Type)
+  return error("`trivial` not defined for type $(type).")
 end
 
-function label_fusion_rule(category_type::Type{<:AbstractCategory}, l1, l2)
-  return error("`label_fusion_rule` not defined for type $(category_type).")
+istrivial(c::AbstractCategory) = (c == trivial(c))
+
+function category_label(c::AbstractCategory)
+  return error("method `category_label` not defined for type $(typeof(c))")
 end
 
-function fusion_rule(c1::AbstractCategory, c2::AbstractCategory)
-  category_type = typeof(c1)
-  return [category_type(l) for l in label_fusion_rule(category_type, label(c1), label(c2))]
+block_dimensions(g::AbstractUnitRange) = block_dimensions(SymmetryStyle(g), g)
+block_dimensions(::AbelianStyle, g) = unlabel.(blocklengths(g))
+function block_dimensions(::NotAbelianStyle, g)
+  return quantum_dimension.(blocklabels(g)) .* blocklengths(g)
+end
+
+quantum_dimension(x) = quantum_dimension(SymmetryStyle(x), x)
+
+function quantum_dimension(::NotAbelianStyle, c::AbstractCategory)
+  return error("method `quantum_dimension` not defined for type $(typeof(c))")
 end
 
+quantum_dimension(::AbelianStyle, ::AbstractCategory) = 1
+quantum_dimension(::AbelianStyle, g::AbstractUnitRange) = length(g)
+quantum_dimension(::NotAbelianStyle, g::AbstractUnitRange) = sum(block_dimensions(g))
+
+# ===============================  Fusion rule interface  ==================================
 ⊗(c1::AbstractCategory, c2::AbstractCategory) = fusion_rule(c1, c2)
 
-⊕(c1::AbstractCategory, c2::AbstractCategory) = [c1, c2]
-⊕(cs::Vector{<:AbstractCategory}, c::AbstractCategory) = [cs; c]
-⊕(c::AbstractCategory, cs::Vector{<:AbstractCategory}) = [c; cs]
+function fusion_rule(c1::AbstractCategory, c2::AbstractCategory)
+  return fusion_rule(combine_styles(SymmetryStyle(c1), SymmetryStyle(c2)), c1, c2)
+end
+
+function fusion_rule(::NotAbelianStyle, c1::C, c2::C) where {C<:AbstractCategory}
+  degen, sectors = label_fusion_rule(C, category_label(c1), category_label(c2))
+  return gradedrange(labelled.(degen, sectors))
+end
+
+# abelian case: return Category
+function fusion_rule(::AbelianStyle, c1::C, c2::C) where {C<:AbstractCategory}
+  _, sectors = label_fusion_rule(C, category_label(c1), category_label(c2))
+  return only(sectors)
+end
```

Comment:
This is kind of a matter of style, but I would prefer:
```julia
function fusion_rule(::AbelianStyle, c1::C, c2::C) where {C<:AbstractCategory}
  return label(only(fusion_rule(NotAbelianStyle(), c1, c2)))
end
```
I realize it involves creating an extra `GradedOneTo` unnecessarily, but I think it makes the intent of the code clearer and the relationship between abelian and non-abelian fusion clearer. It also helps avoid code duplication across the `fusion_rule(::NotAbelianStyle, ...)` and `fusion_rule(::AbelianStyle, ...)`.

If there ends up being a performance issue caused by constructing the intermediate `GradedOneTo`, I think we can optimize that away as needed. For example, we could introduce a type `GradedOneToOne` that is analogous to `OneToOne` that is hardcoded at compile time to only have a single block/sector of size 1 which we could output from `fusion_rule(NotAbelianStyle(), c1, c2)` in abelian cases. I think that's getting ahead of things, though, and I doubt there will be a performance issue with the code I suggested.